invenio_rdm_records/config.py

```diff
@@ -479,3 +479,11 @@ def make_doi(prefix, record):
         ...
     }]
 """
+
+# Feature flag to enable/disable locking file modifications for record with external DOIs on edit record action.
+RDM_LOCK_FILES_FOR_EXTERNAL_DOI = False
+"""Flag to show whether add/delete files of a record on edit action should be locked."""
+
+# Configuration for locking files for external DOIs.
+RDM_EXTERNAL_DOI_PROVIDER_KEY = "external"
```

Comment:
Put this to config, to not have this string hardcoded in app, but it is already hardcoded in many places, as well as the `datacite` provider key. Should use this config variable everywhere instead, WDYT?